### PR TITLE
Optimize indexation

### PR DESCRIPTION
**Pull request by noirbizarre**
This PR optimize indexation by:
- removing some field that weren't in use (ex: `user.role`)
- turning off the automatic derefencing while iterating over objects to only fetch what's necessary

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,7 @@
 - Switch from static theme avatars/placeholders to [identicons](https://en.wikipedia.org/wiki/Identicon) for readability (mostly on discussions) [#1193](https://github.com/opendatateam/udata/pull/1193)
 - Fix the scroll to a discussion sub-thread [#1206](https://github.com/opendatateam/udata/pull/1206)
 - Fix duplication in discussions [migration] [#1209](https://github.com/opendatateam/udata/pull/1209)
+- Optimize a little bit indexation [#1215](https://github.com/opendatateam/udata/pull/1215)
```
Comment (abulte) [inline CHANGELOG.md]: *Optimize indexation a little bit

File: udata/core/dataset/search.py [hunk 2]
```diff
@@ -169,22 +169,25 @@ def is_indexable(cls, dataset):
...
+            organization = Organization.objects(id=dataset.organization.id).first()
+            image_url = organization.logo(40, external=True)
+            owner = None
         elif dataset.owner:
-            image_url = dataset.owner.avatar(40, external=True)
+            owner = User.objects(id=dataset.owner.id).first()
+            image_url = owner.avatar(40, external=True)
+            organization = None
         else:
+            organization = None
```
Comment (abulte) [inline udata/core/dataset/search.py]: Maybe initialize `organization`, `owner` and `image_url` to `None` before the `ifs`? If feel more comfortable with this pattern but YMMV.

File: udata/core/dataset/search.py [hunk 3]
```diff
@@ -169,22 +169,25 @@ def is_indexable(cls, dataset):
...
 
-        certified = dataset.organization and dataset.organization.certified
+        certified = organization and organization.certified
 
         document = {
             'title': dataset.title,
             'description': dataset.description,
-            'license': (dataset.license.id
-                        if dataset.license is not None else None),
+            'license': getattr(dataset.license, 'id', None),
```
Comment (abulte) [inline udata/core/dataset/search.py]: 👌

File: udata/core/reuse/search.py [hunk 4]
```diff
@@ -134,13 +134,23 @@ def serialize(cls, reuse):
...
+        datasets = Dataset.objects(id__in=[r.id for r in reuse.datasets])
+        datasets = list(datasets.only('id', 'title').no_dereference())
+        if reuse.organization:
+            organization = Organization.objects(id=reuse.organization.id).first()
+            owner = None
+        elif reuse.owner:
+            owner = User.objects(id=reuse.owner.id).first()
+            organization = None
+        else:
+            organization = None
```
Comment (abulte) [inline udata/core/reuse/search.py]: Same remark as above about initialization.

File: udata/core/dataset/search.py (hunk 2 above)
Comment (noirbizarre) [reply udata/core/dataset/search.py]: You're right, it's much better

File: udata/core/reuse/search.py (hunk 4 above)
Comment (noirbizarre) [reply udata/core/reuse/search.py]: :+1:

**Pull request merged**